google/cloud/bigquery/job/query.py

```diff
@@ -1430,40 +1430,6 @@ def _reload_query_results(
             timeout=transport_timeout,
         )
 
-    def _done_or_raise(self, retry=DEFAULT_RETRY, timeout=None):
-        """Check if the query has finished running and raise if it's not.
-
-        If the query has finished, also reload the job itself.
-        """
-        # If an explicit timeout is not given, fall back to the transport timeout
-        # stored in _blocking_poll() in the process of polling for job completion.
-        transport_timeout = timeout if timeout is not None else self._transport_timeout
-
-        try:
-            self._reload_query_results(retry=retry, timeout=transport_timeout)
-        except exceptions.GoogleAPIError as exc:
-            # Reloading also updates error details on self, thus no need for an
-            # explicit self.set_exception() call if reloading succeeds.
-            try:
-                self.reload(retry=retry, timeout=transport_timeout)
-            except exceptions.GoogleAPIError:
-                # Use the query results reload exception, as it generally contains
-                # much more useful error information.
-                self.set_exception(exc)
-            finally:
-                return
-
-        # Only reload the job once we know the query is complete.
-        # This will ensure that fields such as the destination table are
-        # correctly populated.
-        if not self._query_results.complete:
-            raise polling_future._OperationNotComplete()
-        else:
-            try:
-                self.reload(retry=retry, timeout=transport_timeout)
-            except exceptions.GoogleAPIError as exc:
-                self.set_exception(exc)
```

Comment:
Thought: We probably should have been calling `set_exception` based on the job status. Need to look into this furteher.

Reply:
OK. We are. 😅 

https://github.com/googleapis/python-bigquery/blob/bf8861c3473a1af978db7a06463ddc0bad86f326/google/cloud/bigquery/job/base.py#L909

Which we call from `_set_properties`

https://github.com/googleapis/python-bigquery/blob/bf8861c3473a1af978db7a06463ddc0bad86f326/google/cloud/bigquery/job/base.py#L677

Which we call from `reload`

https://github.com/googleapis/python-bigquery/blob/bf8861c3473a1af978db7a06463ddc0bad86f326/google/cloud/bigquery/job/base.py#L838